rotkehlchen/exchanges/bybit.py

```diff
@@ -0,0 +1,514 @@
+                )
+
+            if response.status_code != HTTPStatus.OK:
+                raise RemoteError(
+                    f'Bybit api request for {response.url} failed with HTTP status '
+                    f'code {response.status_code} and response {response.text}',
+                )
+
+            break  # else all good, we can break off the retry loop
+
+        log.debug(f'BYBIT RESPONSE: {response.text}')
+        try:
+            json_ret = json.loads(response.text)
+        except json.JSONDecodeError as e:
+            raise RemoteError('Bybit returned invalid JSON response') from e
+
+        if 'result' not in json_ret:
+            raise RemoteError(f'Remote response is missing expected field result: {json_ret}')
+
+        if json_ret['retCode'] != 0 and json_ret['retMsg'] != 'OK':
+            raise RemoteError(f'Bybit returned error in request {json_ret}')
+
+        return json_ret['result']
+
+    def _paginated_api_query(
+            self,
+            endpoint: Literal[
+                'order/history',
+                'asset/deposit/query-record',
+                'asset/withdraw/query-record',
+            ],
+            options: dict[str, Any],
+    ) -> list[dict[str, Any]]:
+        if endpoint == 'order/history':
+            result_key = 'list'
+        else:
+            result_key = 'rows'
+
+        result = []
+        while True:
+            output = self._api_query(path=endpoint, options=options)
+            if (next_cursor := output.get('nextPageCursor')) is not None:
+                options['cursor'] = next_cursor
+
+            result.extend(output[result_key])
+            log.debug(f'Got {len(output[result_key])} items in the query')
+            if len(output[result_key]) < options.get(result_key, PAGINATION_LIMIT):
+                break
+
+        return result
+
+    def query_online_trade_history(
+            self,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> tuple[list[Trade], tuple[Timestamp, Timestamp]]:
+        """
+        Query trades from bybit. This endpoint in the api allows to query by time range
+        """
+        new_trades = []
+        upper_ts, lower_ts = end_ts, Timestamp(end_ts - WEEK_IN_SECONDS)
+        two_years_ago = Timestamp(ts_now() - DAY_IN_SECONDS * 365 * 2)
+        if start_ts < two_years_ago:
+            # bybit allows to query up to two years into the past
+            start_ts = two_years_ago
+
+        while True:
+            options = {
+                'category': 'spot',
+                'endTime': str(ts_sec_to_ms(upper_ts)),
+                'limit': PAGINATION_LIMIT,
+                'startTime': str(ts_sec_to_ms(lower_ts)),
+            }
+            log.debug(f'querying online trades for {options}')
+            raw_data = self._paginated_api_query(endpoint='order/history', options=options)
+            for raw_trade in raw_data:
+                if raw_trade['orderStatus'] != 'Filled':
```

Comment:
This can raise `KeyError`

```suggestion
                if raw_trade.get('orderStatus') != 'Filled':
```